buildpack/layermetadata.go

```diff
@@ -34,23 +36,30 @@ func EncodeLayerMetadataFile(lmf LayerMetadataFile, path, buildpackAPI string) e
 	return errors.New("couldn't find an encoder")
 }
 
-func DecodeLayerMetadataFile(path, buildpackAPI string) (LayerMetadataFile, string, error) { // FIXME: pass the logger and print the warning inside (instead of returning a message)
+func DecodeLayerMetadataFile(path string, buildpackAPI string, logger log.Logger) (LayerMetadataFile, error) { // FIXME: pass the logger and print the warning inside (instead of returning a message)
 	fh, err := os.Open(path)
 	if os.IsNotExist(err) {
-		return LayerMetadataFile{}, "", nil
+		return LayerMetadataFile{}, nil
 	} else if err != nil {
-		return LayerMetadataFile{}, "", err
+		return LayerMetadataFile{}, err
 	}
 	defer fh.Close()
 
 	decoders := supportedEncoderDecoders()
 
 	for _, decoder := range decoders {
 		if decoder.IsSupported(buildpackAPI) {
-			return decoder.Decode(path)
+			lmf, str, err := decoder.Decode(path)
+			if str != "" {
+				logger.Warn(str)
+				if !api.MustParse(buildpackAPI).LessThan("0.6") {
+					return LayerMetadataFile{}, errors.New(str)
+				}
```

Comment:
```suggestion
				if api.MustParse(buildpackAPI).LessThan("0.6") {
					logger.Warn(str)
				} else {
					return LayerMetadataFile{}, errors.New(str)
				}
```
I think this is what we want... the test failures may be due to a test needing to be updated